pkgs/tools/networking/mihomo/default.nix

```diff
@@ -0,0 +1,43 @@
+{ lib
+, fetchFromGitHub
+, buildGoModule
+}:
+buildGoModule rec {
+  pname = "mihomo";
+  version = "1.17.0";
+
+  src = fetchFromGitHub {
+    owner = "MetaCubeX";
+    repo = "mihomo";
+    rev = "v${version}";
+    hash = "sha256-2vkb4fFkms/LuPyUz86T0qCdZgv9W+Db65IGo+6ybLg=";
+  };
+
+  vendorHash = "sha256-/+X2eDCpo8AqWJ7rYbZrYzZapCgrdraTSx6BlWEUd78=";
+
+  # Do not build testing suit
+  excludedPackages = [ "./test" ];
+
+  ldflags = [
+    "-s"
+    "-w"
+    "-X github.com/metacubex/mihomo/constant.Version=${version}"
+  ];
+
+  tags = [
+    "with_gvisor"
+  ];
+
+  # network required
+  doCheck = false;
+
+  meta = with lib; {
+    description = ''
+      A simple Python Pydantic model for Honkai: Star Rail parsed data from the Mihomo API
+    '';
+    homepage = "https://github.com/MetaCubeX/mihomo";
+    license = licenses.mit;
+    maintainers = with maintainers; [ oluceps ];
+    mainProgram = "mihomo";
```

Comment:
> Can we provide an alias for its original name `clash-meta`? This may enhance backward compatibility for legacy third-party programs or scripts.

Alias added, attached a warning.